src/game/server/gamecontext.cpp

```diff
@@ -129,28 +129,109 @@ class CCharacter *CGameContext::GetPlayerChar(int ClientID)
 	return m_apPlayers[ClientID]->GetCharacter();
 }
 
-int CGameContext::GetZombieCount() {
-	int count = 0;
+int CGameContext::GetActivePlayerCount(){
+	
+	// returns how many players are currently playing and not spectating
+		int PlayerCount = 0;
+		int SpecCount = 0;
+		for(int i=0; i<MAX_CLIENTS; i++)
+		{
+			//COMPILER ERROR ‘class IServer’ has no member named ‘m_aClients’
+			if(Server()->m_aClients[i].m_State == CServer::CClient::STATE_INGAME)
+			{
+				if (m_apPlayers[i]->GetTeam() != TEAM_SPECTATORS)
+					PlayerCount++;
+				else 
+					SpecCount++;
+			}
+		}
+		
+		m_NbActivePlayers = PlayerCount;
+		m_NbSpectators = SpecCount;
+
+		dbg_msg("infclass", "Active Players: %d -- Spectators: %d", m_NbActivePlayers, m_NbSpectators);
+		
+		return PlayerCount;
+}
+
+int CGameContext::GetSpectatorCount(){
+	
+	int PlayerCount = 0;
+	int SpecCount = 0;
+	for(int i=0; i<MAX_CLIENTS; i++)
+	{
+		
+		//COMPILER ERROR ‘class IServer’ has no member named ‘m_aClients’
+		if(Server()->m_aClients[i].m_State == CServer::CClient::STATE_INGAME && m_apPlayers[i]->GetTeam() == TEAM_SPECTATORS)
+			SpecCount++;
+		else 
+			PlayerCount++;
+	}
+	
+	m_NbActivePlayers = PlayerCount;
+	m_NbSpectators = SpecCount;
+	
+	return SpecCount;
+}
+
+int CGameContext::GetHumanCount(){
+	
+	int humanCounter = 0;
+	int zombieCounter = 0;
+	for(int i = 0; i < MAX_CLIENTS; i++)
+	{
+		if (!m_apPlayers[i])
+			continue;
+		if (m_apPlayers[i]->IsHuman())
+			humanCounter++;
+		else
+			zombieCounter++;
+	}
+	m_NbHumans = humanCounter;
+	m_NbZombies = zombieCounter;
+	
+	dbg_msg("infclass", "#humans: %d -- #zombies: %d", m_NbHumans, m_NbZombies);
+	
+	return humanCounter;
+	
+}
+
+int CGameContext::GetZombieCount(){
+	
+	int humanCounter = 0;
+	int zombieCounter = 0;
 	for(int i = 0; i < MAX_CLIENTS; i++)
 	{
 		if (!m_apPlayers[i])
 			continue;
 		if (m_apPlayers[i]->IsZombie())
-			count++;
+			zombieCounter++;
+		else
+			humanCounter++;
 	}
-	return count;
+	
+	m_NbZombies = zombieCounter;
+	m_NbHumans = humanCounter;
+	
+	dbg_msg("infclass", "#humans: %d -- #zombies: %d", m_NbHumans, m_NbZombies);
+	
+	return zombieCounter;
+	
 }
 
-int CGameContext::GetZombieCount(int zombie_class) {
+int CGameContext::GetIsOfClassCount(int player_class) {
+	
 	int count = 0;
 	for(int i = 0; i < MAX_CLIENTS; i++)
 	{
 		if (!m_apPlayers[i])
 			continue;
-		if (m_apPlayers[i]->IsZombie() && m_apPlayers[i]->GetClass() == zombie_class)
+		if (m_apPlayers[i]->IsZombie() && m_apPlayers[i]->GetClass() == player_class)
 			count++;
 	}
 	return count;
+	
+	
```

Comment:
GetIsOfClassCount -> maybe name it GetZombieClassCount ? 
Why change the parameter name from zombie_class to player_class?
Or do you want to count human classes as well with it? Right now it can only count zombie classes

Reply:
I make it more general. You are right.